src/Transformers/ModelTransformer.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Spatie\LaravelTypeScriptTransformer\Transformers;
+
+use Spatie\TypeScriptTransformer\Structures\TransformedType;
+
+use DB;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Schema;
+use ReflectionClass;
+
+/**
+ * A Transformer that generates Typescript definitions for a Model by inspecting the DB Schema
+ */
+class ModelTransformer implements Transformer
+{
+    public function transform(ReflectionClass $class, string $name): ?TransformedType
+    {
+        if (!is_subclass_of($class->name, Model::class)) {
+            return null;
+        }
+
+        /** @var Model $modelInstance */
+        $modelInstance = $class->newInstanceWithoutConstructor();
+
+        $table = $modelInstance->getTable();
+        $columns = Schema::getColumnListing($table);
+
+        $hidden = $modelInstance->getHidden();
+
+        $serializedColumns = array_diff($columns, $hidden);
+
+        $column_defs = collect(DB::select("
+            SELECT column_name, is_nullable, data_type
+            FROM information_schema.columns
+            WHERE table_name = '$table';
+        "));
```

Comment:
Note: I wrote this transformer a few weeks ago, so I can't remember why I went with this approach. I was likely having trouble getting the data types using the `Schema` facade. 

I am not _certain_, but I am fairly confident that this is not an appropriate solution. This was written against a PostgreSQL database. It seems highly unlikely to me that it would work for other DBs.